### PR TITLE
improve build setting

### PR DESCRIPTION
**Pull request by kotarella1110**
- improve rollup setting
    - change output filename
        - `dist/resolver.js` to `dist/index.js`
        - `dist/resolver.es.js` to `dist/index.esm.js`
    - support umd build
        - `dist/index.umd.min.js`
        - umd name is `ReactHookFormResolvers`
        - https://unpkg.com/@hookform/resolvers@0.0.4/dist/index.umd.min.js
        - https://cdn.jsdelivr.net/npm/@hookform/resolvers@0.0.4/dist/index.umd.min.js
- improve eslint & prettier setting
- improve package.json
    - move react-hook-form from deps to peerDeps
    - add husky and lint-staged
- improve actions workflow
- improve types #12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,52 +1,97 @@
...
+    "scheme",
+    "validation",
+    "scheme-validation",
+    "hookform",
+    "react-hook-form",
+    "yup",
+    "joi",
+    "sperstruct",
+    "typescript"
+  ],
```
Comment (kotarella1110) [inline package.json:39]: @bluebill1049 Is this keyword OK?

File: rollup.config.js [hunk 2]
```diff
@@ -1,39 +1,59 @@
...
-      exports: 'named',
-    },
-    {
-      file: `dist/resolvers.es.js`,
-      format: 'esm',
+export default [
+  {
+    input: 'src/index.ts',
+    output: {
+      name: 'HookFormResolvers',
```
Comment (kotarella1110) [inline rollup.config.js:14]: @bluebill1049 Is this umd name OK?
`HookFormResolvers` vs `ReactHookFormResolvers` vs `Resolvers`
please see [umd example](https://codesandbox.io/s/react-hook-form-umd-n8z43?file=/index.html).

**Comment by kotarella1110** [comment]
@bluebill1049 I will fix @hookform/devtools  as well.

File: .npmignore [hunk 3]
```diff
@@ -1,39 +0,0 @@
-example/
```
Comment (bluebill1049) [inline .npmignore:1]: why remove this?

File: .prettierrc [hunk 4]
```diff
@@ -1,4 +0,0 @@
-{
```
Comment (bluebill1049) [inline .prettierrc:1]: ?

File: README.md [hunk 5]
```diff
@@ -14,7 +14,8 @@
 [![npm](https://img.shields.io/npm/dt/@hookform/resolvers.svg?style=for-the-badge)](https://www.npmjs.com/package/@hookform/resolvers)
 [![npm](https://img.shields.io/bundlephobia/minzip/@hookform/resolvers?style=for-the-badge)](https://bundlephobia.com/result?p=@hookform/resolvers)
 
-[![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social)](https://twitter.com/intent/tweet?text=React+hooks+for+form+validation+without+the+hassle&url=https://github.com/bluebill1049/@hookform/resolvers)&nbsp;[![Join the community on Spectrum](https://withspectrum.github.io/badge/badge.svg)](https://spectrum.chat/react-hook-form)
+[![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social)](https://twitter.com/intent/tweet?text=React+hooks+for+form+validation+without+the+hassle&url=https://github.com/react-hook-form/resolvers)
+[![Join the community on Spectrum](https://withspectrum.github.io/badge/badge.svg)](https://spectrum.chat/react-hook-form)
```
Comment (bluebill1049) [inline README.md:18]:
```suggestion
```

I think going forward we are leaner toward github dicussion.

File: README.md [hunk 6]
```diff
@@ -14,7 +14,8 @@
 [![npm](https://img.shields.io/npm/dt/@hookform/resolvers.svg?style=for-the-badge)](https://www.npmjs.com/package/@hookform/resolvers)
 [![npm](https://img.shields.io/bundlephobia/minzip/@hookform/resolvers?style=for-the-badge)](https://bundlephobia.com/result?p=@hookform/resolvers)
 
-[![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social)](https://twitter.com/intent/tweet?text=React+hooks+for+form+validation+without+the+hassle&url=https://github.com/bluebill1049/@hookform/resolvers)&nbsp;[![Join the community on Spectrum](https://withspectrum.github.io/badge/badge.svg)](https://spectrum.chat/react-hook-form)
+[![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social)](https://twitter.com/intent/tweet?text=React+hooks+for+form+validation+without+the+hassle&url=https://github.com/react-hook-form/resolvers)
```
Comment (bluebill1049) [inline README.md:17]:
```suggestion
[![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social)](https://twitter.com/intent/tweet?text=React+hooks+for+form+validation+without+the+hassle&url=https://github.com/react-hook-form/resolvers)
```

don't think anyone use it. let's remove it as well keep it clean and minimum.

File: package.json (hunk 1 above)
Comment (bluebill1049) [reply package.json:39]: awesome! thanks

File: rollup.config.js (hunk 2 above)
Comment (bluebill1049) [reply rollup.config.js:14]: `ReactHookFormResolvers` 👍 what do you think? looks better at umd example.

File: .npmignore (hunk 3 above)
Comment (kotarella1110) [reply .npmignore:1]: I fixed to use `files`, we can remove the `.npmignore`.

https://docs.npmjs.com/files/package.json#files

https://github.com/react-hook-form/resolvers/pull/11/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R13-R15

Comment (bluebill1049) [reply .npmignore:1]: can you do the same for react-hook-form?

File: .prettierrc (hunk 4 above)
Comment (kotarella1110) [reply .prettierrc:1]: I moved prettier setting to `.eslintrc.js`.
I think it's more useful to be able to manage with one file.
What do you think?

https://github.com/react-hook-form/resolvers/blob/57795af37f395d4147ba5ee76c782f47afc247f5/.eslintrc.js#L15-L21

File: README.md (hunk 5 above)
Comment (kotarella1110) [reply README.md]: done 👍

File: README.md (hunk 6 above)
Comment (kotarella1110) [reply README.md]: done 👍

File: .npmignore (hunk 3 above)
Comment (kotarella1110) [reply .npmignore:1]: yea! I will fix 👍

File: .prettierrc (hunk 4 above)
Comment (bluebill1049) [reply .prettierrc:1]: personally, I prefer to keep them separate. They are two different technology, then others may ask why not keep `eslint` inside `prettier`.

File: rollup.config.js (hunk 2 above)
Comment (kotarella1110) [reply rollup.config.js:14]: we may use `HookForm` as the umd name of react-hook-form.
Which is better?

* HookForm
* HookFormResolvers
* HookFormErrorMessage
* HookFormDevtools

vs

* ReactHookForm
* ReactHookFormResolvers
* ReactHookFormErrorMessage
* ReactHookFormDevtools

Comment (bluebill1049) [reply rollup.config.js:14]: yea better! first one is better!

Comment (bluebill1049) [reply rollup.config.js:14]: actually maybe second one, what if one day we have angularHookForm? lol

File: .prettierrc (hunk 4 above)
Comment (kotarella1110) [reply .prettierrc:1]: done.

File: rollup.config.js (hunk 2 above)
Comment (kotarella1110) [reply rollup.config.js:14]: Hahaha. if it has the potential to support Angular and Vue, I think `ReactHookForm` is better.
However, in that case we think it is necessary to reconsider the scope name (`@hookform`).
ex: `@hookform` to `@react-hook-form`

Comment (bluebill1049) [reply rollup.config.js:14]: @hookform/vue is fine as well?

if we can change to @hookform/react later on.

Comment (kotarella1110) [reply rollup.config.js:14]: awesome! I will change umd name from `HookForm` to `ReactHookForm`.

Comment (kotarella1110) [reply rollup.config.js]: done.

**Comment by kotarella1110** [comment]
Would you also like to fix the output name of react-hook-form? Or do you want to leave it as is?

https://github.com/react-hook-form/react-hook-form/blob/master/package.json#L5-L6

**Comment by bluebill1049** [comment]
> Would you also like to fix the output name of react-hook-form? Or do you want to leave it as is?
> 
> * `dist/react-hook-form.js` to `dist/index.js`
> * `dist/react-hook-form.es.js` to `dist/index.esm.js`
> * `dist/react-hook-form.umd.js` to `dist/index.umd.js`
> * `dist/react-hook-form.ie11.js` to `dist/index.ie11.js`
> 
> https://github.com/react-hook-form/react-hook-form/blob/0fee0da0044e1cd3afa887fa046dd84e92e4ed44/package.json#L5-L6

sounds good to me. as long as we can do a quick test, you can still easily find the source code under chrome developer tab.

**Comment by kotarella1110** [comment]
I will fix it 👍

**Comment by kotarella1110** [comment]
This PR includes a version release, which is better, squash merge or merge commit?

**Comment by bluebill1049** [comment]
> This PR includes a version release, which is better, squash merge or merge commit?

I normally run `npm version` at master and push the tag into master, but whatever works.

**Comment by kotarella1110** [comment]
@bluebill1049 "Create a merge commit" OK?

スクリーンショット 2020-06-16 11 32 56

**Comment by bluebill1049** [comment]
squash merge, please :)

**Pull request merged**